Merge_Sort/Merge_Sort.rs

```diff
@@ -0,0 +1,65 @@
+use std::io;
+use std::str::FromStr;
+
+fn main() {
+    println!("Enter the numbers to be sorted (separated by space) : ");
+    let mut i = read_values::<i32>().unwrap();
+    merge_sort(&mut i);
+    println!("Sorted array: {:?}", i)
+    }
+
+fn merge_sort(a: &mut Vec<i32>) {
+    let size = a.len();
+    let mut buffer = vec![0; size];
+    merge_split(a, 0, size, &mut buffer);
+}
+
+fn merge_split(a: &mut Vec<i32>, start: usize, end: usize, buffer: &mut Vec<i32>) {
+    if end - start > 1 {
+        // Determine the split index
+        let split = (start + end) / 2;
+		
+		// Split and merge sort
+        merge_split(a, start, split, buffer);
+        merge_split(a, split, end, buffer);
+        merge(a, buffer, start, split, end);
+       merge_copy(buffer, a, start, end);
```

Comment:
Indentation is not consistent here. Please ensure 4 space indentation inside the if block.